src/fermilib/utils/_operator_utils.py

```diff
@@ -151,19 +150,17 @@ def save_operator(operator, file_name=None, data_directory=None):
     else:
         raise TypeError('Operator of invalid type.')
 
-    with h5py.File(file_path, 'w') as f:
-        f['operator_type'] = numpy.string_(operator_type)
-        f['operator_terms'] = numpy.string_(json.dumps(
-            [{'k': k, 'r': v.real, 'i': v.imag}
-                for k, v in operator.terms.items()],
-            ensure_ascii=False).encode('utf8'))
+    tm = operator.terms
+    f = open(file_path, 'wb')
+    marshal.dump((operator_type, {k: numpy_scalar(tm[k]) for k in tm}), f)
```

Comment:
In my opinion, the coefficients of both FermionOperator and QubitOperator should always be of native Python complex type when the operators are originally constructed, so that we don't need to apply the transformations/castings at all here. Is it possible to make that change in FermionOperator and QubitOperator? 
If not, I agree that casting to complex directly here is cleaner.
@idk3 @babbush @jarrodmcc